languages/rust/generic/Parse_rust_tree_sitter.ml

```diff
@@ -1194,11 +1194,14 @@ and map_expression (env : env) (x : CST.expression) =
       in
       let rhs = map_expression env v3 in
       G.AssignOp (lhs, (op, tok), rhs)
-  | `Type_cast_exp (v1, v2, v3) ->
+  | `Type_cast_exp (v1, v2, v3) -> (
```

Comment:
One thing I'm slightly afraid of is the fact that this takes away the possibility of matching an explicit cast within Rust, at least by using a metavariable. For instance, suppose I wanted to find cases of casting a variable to `int32`, I would probably want to write the pattern:
```
$X as int32
```
but this will now match as a typed metavariable, and match many things which are not just an explicit cast. I'd like it if we could preserve the choice, to the rule-writer, of which they want to match.

Additionally, this is a bit of a tough case, because I think the language of `as` lends itself to more thinking of code where you have explicitly said, "transform this expression into one of this type", versus the typing in parameters (like `x : t`), which is more passive (it reflects a property, that `x` has type `t`). In an ideal world, I'd prefer to write a pattern like `$X : ty` in Rust, to avoid that. We could do this by modifying the `tree-sitter` Semgrep grammar for Rust, but that might be a lot of work, so we don't have to do that.

At the very least, I notice that usually we surround Semgrep typed metavariables with parentheses. Could we move this case under `Paren_exp`, down below, so that we continue this behavior (and we allow some measure of using patterns as casts instead of typed metavariables)?

I will consult about the `$X : ty` thing in the meantime.

Reply:
It seems that we have consensus on preferring `$X : ty` as the syntax for typed metavariables, as per this thread: https://semgrepinc.slack.com/archives/C01NXGX2EHZ/p1688587347895529

Could you make that change? You should have to touch the `grammar.js` from `semgrep-rust` within `ocaml-tree-sitter-semgrep`, and it will entail making a PR to `ocaml-tree-sitter-semgrep` with the updated grammar, and then a PR to `semgrep` updating our version of `semgrep-rust`.